scripts/india_nhm/base/data_cleaner.py

```diff
@@ -0,0 +1,227 @@
+                cleaned_df['Date'] = date
+
+                # If no columns specified, extract all except first two (index and state name)
+                if not self.cols_to_extract:
+                    self.cols_to_extract = list(
+                        set(self.raw_df.columns.droplevel(-1)[2:]))
+
+                # Extract specified columns from raw dataframe if it exists
+                for col in self.cols_to_extract:
+                    if col in self.raw_df.columns:
+                        cleaned_df[col] = self.raw_df[col].iloc[1:]
+                    else:
+                        continue
+
+                df_full = df_full.append(cleaned_df, ignore_index=True)
+                
+             
+        # Converting column names according to schema and saving it as csv
+        df_full['DistrictCode'] = df_full.apply(lambda row: self._get_district_code(row),
+                                                axis=1)
+        df_full = df_full.groupby(level=0, axis=1, sort=False).sum()
+        df_full.columns = df_full.columns.map(self.cols_dict)
+
+        df_full.iloc[2:].to_csv(self.final_csv_path, index=False)
+
+
+    def create_mcf_tmcf(self):
+        """
+        Class method to generate MCF and TMCF files for the current dataset.
+        
+        """
+        tmcf_file = "{}.tmcf".format(self.dataset_name)
+        mcf_file = "{}.mcf".format(self.dataset_name)
+
+        with open(tmcf_file, 'w+') as tmcf, open(mcf_file, 'w+') as mcf:
+            # Writing isoCODE entity
+            tmcf.write(TMCF_ISOCODE.format(dataset_name=self.dataset_name))
+
+            # Keeping track of written StatVars
+            # Some columns in NHM_FamilyPlanning have same StatVar for multiple cols
+            statvars_written = []
+
+            # Writing nodes for each StatVar
+            for idx, variable in enumerate(self.cols_to_extract):
+                if self.cols_dict[variable] not in statvars_written:
+                    # Writing TMCF
+                    tmcf.write(
+                        TMCF_NODES.format(dataset_name=self.dataset_name,
+                                          index=idx + 1,
+                                          statvar=self.cols_dict[variable]))
+                    # Writing MCF
+                    mcf.write(
+                        MCF_NODES.format(
+                            statvar=self.cols_dict[variable],
+                            description=self.clean_names[variable]))
+
+                    statvars_written.append(self.cols_dict[variable])
+                    
+    def _get_district_code(self, row):
+        if pd.notna(row['District']):
+            close_match = difflib.get_close_matches(row['District'].upper(), self.dist_code['DistrictName(InEnglish)'],
+                                n=1, cutoff=0.8)
+            if close_match:
+                return self.dist_code[self.dist_code['DistrictName(InEnglish)'] == close_match[0]]['DistrictCode'].values[0]
```

Comment:
same question as in the other covid19 districts PR about efficacy

Reply:
Added some comments here too.